app/src/test/java/ayds/lisboa/songinfo/moredetails/model/repository/CardRepositoryTest.kt

```diff
@@ -0,0 +1,87 @@
+package ayds.lisboa.songinfo.moredetails.model.repository
+
+import ayds.lisboa.songinfo.otherdetails.model.entities.ServiceCard
+import ayds.lisboa.songinfo.otherdetails.model.entities.Source
+import ayds.lisboa.songinfo.otherdetails.model.repository.CardRepository
+import ayds.lisboa.songinfo.otherdetails.model.repository.CardRepositoryImpl
+import ayds.lisboa.songinfo.otherdetails.model.repository.external.Broker
+import ayds.lisboa.songinfo.otherdetails.model.repository.external.ProxyLastFM
+import ayds.lisboa.songinfo.otherdetails.model.repository.local.service.CardLocalStorage
+import ayds.lisboa1.lastfm.LastFMArtistBiography
+import ayds.lisboa1.lastfm.LastFMService
+import io.mockk.every
+import org.junit.Assert.*
+import io.mockk.mockk
+import io.mockk.verify
+import org.junit.Test
+
+class CardRepositoryTest {
+
+    private val cardLocalStorage: CardLocalStorage = mockk(relaxUnitFun = true)
+    private val broker: Broker = mockk(relaxUnitFun = true)
+
+    private val cardRepository: CardRepository by lazy {
+        CardRepositoryImpl(cardLocalStorage, broker)
+    }
+
+    @Test
+    fun `given existing artistBiography by name should return artistBiography and mark it as local`() {
+        val cardList: List<ServiceCard> = listOf(
+            ServiceCard("artist", "description", "infoUrl", Source.LASTFM, "sourceLogoUrl", false)
+        )
+
+        every { cardLocalStorage.getInfo("artist") } returns cardList
+
+        val result = cardRepository.getArtistInfo("artist")
+
+        assertEquals(cardList, result)
+        assertTrue(cardList.first().isLocallyStored)
```

Comment:
Debería checkear que todos los elementos tengan la propiedad true (de manera más genérica)
```suggestion
        assertTrue(cardList.all { it.isLocallyStored })
```